benchmarks/IO.py

```diff
@@ -28,22 +40,37 @@ def time_ak_write(N_per_locale, numfiles, trials, dtype, path, seed, parquet, co
         a = ak.randint(0, 1, N, dtype=ak.float64, seed=seed)
     elif dtype == "uint64":
         a = ak.randint(0, 2**32, N, dtype=ak.uint64, seed=seed)
+    elif dtype == "str":
+        a = ak.random_strings_uniform(1, 16, N, seed=seed)
 
-    writetimes = []
-    for i in range(trials):
-        for j in range(numfiles):
-            start = time.time()
-            a.to_hdf(f"{path}{j:04}") if not parquet else a.to_parquet(
-                f"{path}{j:04}", compression="snappy" if compressed else None
-            )
-            end = time.time()
-            writetimes.append(end - start)
-    avgwrite = sum(writetimes) / trials
-
-    print("write Average time = {:.4f} sec".format(avgwrite))
+    times = {}
+    if parquet:
+        for comp in comps:
+            if comp in COMPRESSIONS:
+                writetimes = []
+                for i in range(trials):
+                    for j in range(numfiles):
+                        start = time.time()
+                        a.to_parquet(
+                            f"{path}{comp}{j:04}", compression=None if comp == "none" else comp
+                        )
+                        end = time.time()
+                        writetimes.append(end - start)
+                times[comp] = sum(writetimes) / trials
+    else:
+        writetimes = []
+        for i in range(trials):
+            for j in range(numfiles):
+                start = time.time()
+                a.to_hdf(f"{path}{j:04}")
+                end = time.time()
+                writetimes.append(end - start)
+        times["HDF5"] = sum(writetimes) / trials
 
     nb = a.size * a.itemsize * numfiles
-    print("write Average rate = {:.2f} GiB/sec".format(nb / 2**30 / avgwrite))
+    for key in times.keys():
+        print("write Average time {} = {:.4f} sec".format(key, times[key]))
+        print("write Average rate {} = {:.2f} GiB/sec".format(key, nb / 2**30 / times[key]))
 
 
 def time_ak_read(N_per_locale, numfiles, trials, dtype, path, seed, parquet):
```

Comment:
Question for the reads, we don't do any iteration on the compression here. So I'm not familiar enough with this workflow to say definitively (maybe @Ethan-DeBandi99 can chime in), but it seems like it would read all the different file compressions and average them together maybe?

Before when doing the full read write workflow, you were either h5, pq with compression, or pq without compression. So the read was always measuring exactly what the write was doing. Now it feels like we could have a couple different write times broken out by compression, but only one read time which could vary quite a lot depending which compressions the files were written with

I'm not sure if this is worth changing but I def wanted to capture the discussion around it

Reply:
This is now implemented